src/contracts/interfaces/IAllocationManager.sol

```diff
@@ -0,0 +1,330 @@
+     */
+    function clearDeallocationQueue(
+        address operator,
+        IStrategy[] calldata strategies,
+        uint16[] calldata numToClear
+    ) external;
+
+    /**
+     * @notice Called by the delegation manager to set an operator's allocation delay.
+     * This is set when the operator first registers, and is the time between an operator
+     * allocating magnitude to an operator set, and the magnitude becoming slashable.
+     * @param operator The operator to set the delay on behalf of.
+     * @param delay the allocation delay in seconds
+     */
+    function setAllocationDelay(address operator, uint32 delay) external;
+
+    /**
+     * @notice Called by an operator to set their allocation delay. This is the time between an operator
+     * allocating magnitude to an operator set, and the magnitude becoming slashable.
+     * @dev Note that if an operator's allocation delay is 0, it has not been set yet,
+     * and the operator will be unable to allocate magnitude to any operator set.
+     * @param delay the allocation delay in seconds
+     */
+    function setAllocationDelay(
+        uint32 delay
+    ) external;
+
+    /**
+     *
+     *                         VIEW FUNCTIONS
+     *
+     */
+
+    /**
+     * @notice Returns the effective magnitude info for each of an operator's operator sets.
+     * This method fetches the complete list of an operator's operator sets, then applies any
+     * completable allocation modifications to return the effective, up-to-date current and
+     * pending magnitude allocations for each operator set.
+     * @param operator the operator to query
+     * @param strategy the strategy to get allocation info for
+     * @return the list of the operator's operator sets
+     * @return the corresponding allocation details for each operator set
+     */
+    function getAllocationInfo(
+        address operator,
+        IStrategy strategy
+    ) external view returns (OperatorSet[] memory, MagnitudeInfo[] memory);
+
+    /**
+     * @notice Returns the effective magnitude info for each operator set. This method
+     * automatically applies any completable modifications, returning the effective
+     * current and pending allocations for each operator set.
+     * @param operator the operator to query
+     * @param strategy the strategy to get allocation info for
+     * @param operatorSets the operatorSets to get allocation info for
+     * @return The magnitude info for each operator set
+     */
+    function getAllocationInfo(
+        address operator,
+        IStrategy strategy,
+        OperatorSet[] calldata operatorSets
+    ) external view returns (MagnitudeInfo[] memory);
+
+    /**
+     * @notice Returns the effective magnitude info for each operator for each strategy for the operatorSet This method
+     * automatically applies any completable modifications, returning the effective
+     * current and pending allocations for each operator set.
```

Comment:
How to differentiate between current and pending allocations? Source code seems to fetch only pending allocations:
```solidity
    /// @inheritdoc IAllocationManager
    function getAllocationInfo(
        OperatorSet calldata operatorSet,
        IStrategy[] calldata strategies,
        address[] calldata operators
    ) public view returns (MagnitudeInfo[][] memory) {
        MagnitudeInfo[][] memory infos = new MagnitudeInfo[][](operators.length);
        for (uint256 i = 0; i < operators.length; ++i) {
            for (uint256 j = 0; j < strategies.length; ++j) {
                PendingMagnitudeInfo memory info = _getPendingMagnitudeInfo({
                    operator: operators[i],
                    strategy: strategies[j],
                    operatorSetKey: _encodeOperatorSet(operatorSet)
                });

                infos[i][j] = MagnitudeInfo({
                    currentMagnitude: info.currentMagnitude,
                    pendingDiff: info.pendingDiff,
                    effectTimestamp: info.effectTimestamp
                });
            }
        }

        return infos;
    }
```
What am I missing? Does it have to do with `currentMagnitude` and `pendingDiff` fields?

Reply:
I've refactored this in https://github.com/Layr-Labs/eigenlayer-contracts/pull/860 to be a bit more clear, but the *thing being returned is actually the current/most up to date state.*

The `_getPendingMagnitudeInfo` naming is meant to imply that the changes should be written to state if they're meant to persist. But in this case we're just reading them so it's a bit unclear.